Pod/Classes/ResultType.swift

```diff
@@ -0,0 +1,123 @@
+    associatedtype Value
+    
+    init(_ value: Value?)
+    init(_ error: Error)
+    
+    var value: Value? { get }
+    var error: Error? { get }
+}
+
+public enum Result<T>: ResultType {
+    public typealias Value = T
+    
+    case success(Value?)
+    case failure(Error)
+    
+    public var value: Value? {
+        switch self {
+        case .success(let v): return v
+        default: return nil
+        }
+    }
+    
+    public var error: Error? {
+        switch self {
+        case .failure(let e): return e
+        default: return nil
+        }
+    }
+    
+    public init(_ error: Error) {
+        self = .failure(error)
+    }
+    
+    public init(_ value: Value?) {
+        if let v = value as? Error {
+            self = .failure(v)
+        }
+        else if let v = value {
+            self = .success(v)
+        }
+        else {
+            self = .success(nil)
+        }
+    }
+}
+
+
+public extension Result {
+    public var isEmpty: Bool {
+        return value == nil
+    }
+}
+
+// MARK: - Helper Handling -
+public extension Result {
+    func onSuccess(handler: (_ data: Value?) -> Void) {
+        switch(self) {
+        case .success(let data): handler(data)
+        default: break
+        }
+    }
+    
+    func onError(handler: (_ err: Error) -> Void) {
+        switch(self) {
+        case .failure(let e): handler(e)
+        default: break
+        }
+    }
+}
+
+public extension Result where T:Equatable {
+    public static func ==(lhs: Result, rhs: Result) -> Bool {
+        switch (lhs, rhs) {
+        case (.success(let lhsVal), .success(let rhsVal)):
+            // Two successes are equal if their internal vals are equal
+            return lhsVal == rhsVal
+            
+        case (.failure(let lhsErr), .failure(let rhsErr)):
+            // Two errors are equal if their _domain's and _code's are equal
+            return lhsErr._domain == rhsErr._domain && lhsErr._code == rhsErr._code
+            
+        default: return false
+        }
+    }
+}
+
+// MARK: - Equatability -
+extension Result: Equatable {}
+@objc public class NoResponseType: NSObject {}
+
+public func ==<T,U>(lhs: Result<T>, rhs: Result<U>) -> Bool {
+    // Results with different types are not equal
+    switch (lhs, rhs) {
+    case (.failure(let lhsErr), .failure(let rhsErr)):
+        // Two errors are equal if their _domain's and _code's are equal
+        return lhsErr._domain == rhsErr._domain && lhsErr._code == rhsErr._code
+        
+    default: return false
+    }
+}
+
+public func ==<T>(lhs: Result<T>, rhs: Result<T>) -> Bool {
+    switch (lhs, rhs) {
+    case (.success, .success) where T.self == NoResponseType.self:
+        return true
+    case (.failure(let lhsErr), .failure(let rhsErr)):
+        // Two errors are equal if their _domain's and _code's are equal
+        return lhsErr._domain == rhsErr._domain && lhsErr._code == rhsErr._code
+        
+    default: return false
+    }
+}
```

Comment:
Moved over the tests from Router.  Will open a PR for router to remove ResultType and ResultTypeTests from URBNRouter